src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -210,48 +200,31 @@ public void stopIntake() {
         intake.set(TalonSRXControlMode.PercentOutput, 0);
     }
 
+    public boolean checkDependencies() {
+        return checkHoopVision() && checkChassis();
+    }
+
     // Validate hoop vision
     public boolean checkHoopVision() {
         return FeatureFlags.doVision && FeatureFlags.visionInitialized && vc.isHoopValid();
     }
 
-    public void autoShoot() {
-        vc.autoShoot();
+    public boolean checkChassis() {
+        return FeatureFlags.doChassis && FeatureFlags.chassisInitialized;
     }
 
-    private void updateManualRPMS() {
-        double rpmOld = shooterRpms;
-        if (oi.copilot.getDPadPress(oi.DPadUp)) {
-            System.out.println("increasing");
-            shooterRpms += 100;
-        }
-
-        else if (oi.copilot.getDPadPress(oi.DPadDown)) {
-            System.out.println("decrease");
-            shooterRpms -= 100;
-        }
-
-        if (shooterRpms < 0) {
-            shooterRpms = 0;
-        }
-
-        else if (shooterRpms > 15000) {
-            shooterRpms = 15000;
-        }
-
-        if (rpmOld != shooterRpms) {
-            fl.write(Double.toString(shooterRpms));
-        }
+    public void autoShoot() {
+        vc.autoShoot();
     }
 
     public double calculateShooterRPMS(double distance) {
         // distance = hoopx
         double shooterRPM = 0;
-        final double angle = 45;
-        final double diameter = 0.5;
+        final double angle = 45; // angle in degrees
+        final double diameter = 6; // distance in inches
         double velocity = 0;
         // math
-        velocity = Math.sqrt(distance * 9.8 / Math.toDegrees(Math.sin(Math.toRadians(2 * angle))));
+        velocity = Math.sqrt(distance * 12 * 9.8 / Math.toDegrees(Math.sin(Math.toRadians(2 * angle))));
```

Comment:
This whole function needs to be replaced by an empirical one determined from our calibration data. For now, I would go with something like
```
2000 + 1000 * (distance - 100) / 120
```

assuming our earlier test hit
```
100" at 2000 RPM
220" at 3000 RPM
```
(which may or may not be correct, I don't have the data that was recorded).

The actual function is probably not linear; we will fit a curve to the full set of calibration data we collect once we have an opportunity to test the shooter for real.

Reply:
agreed